sdk/python/feast/infra/online_stores/remote.py

```diff
@@ -0,0 +1,180 @@
+            while feature_value_index < len(requested_features):
+                feature_values_dict = dict()
+                for index, feature_name in enumerate(response_json['metadata']['feature_names']):
+                    if self._check_if_feature_requested(feature_name, requested_features):                                                
+                        message = python_values_to_proto_values([response_json['results'][index]['values'][
+                            feature_value_index]], ValueType.UNKNOWN)
+                        feature_values_dict[feature_name] = message[0]                                                
+                result_tuples.append((event_ts, feature_values_dict))
+                feature_value_index += 1
+            return result_tuples
+        else:
+            error_msg = f"Unable to retrieve the online store data using feature server API. Error_code={response.status_code}, error_message={response.reason}"
+            logger.error(error_msg)
+            raise RuntimeError(error_msg)
+
+    def _construct_online_read_api_json_request(self, entity_keys: List[EntityKeyProto], table: FeatureView, requested_features: Optional[List[str]] = None):
+        api_requested_features = []
+        for requested_feature in requested_features:
+            api_requested_features.append(f"{table.name}:{requested_feature}")
+
+        entity_values = []
+        entity_key = ""
+        for row in entity_keys:
+            entity_key = row.join_keys[0]
+            entity_values.append(getattr(row.entity_values[0], row.entity_values[0].WhichOneof('val')))
+
+        req_body = json.dumps(
+            {"features": api_requested_features, "entities": {entity_key: entity_values}}
+        )
+        return req_body
+    
+    def _construct_push_api_json_request(self, data: Tuple[EntityKeyProto, Dict[str, ValueProto], datetime, Optional[datetime]
+            ], table: FeatureView):            
+        json_df = {}
+        
+        entity_values = []
+        entity_key = data[0].join_keys[0]
+        for row in data[0].entity_values:
+            entity_values.append(getattr(row, row.WhichOneof('val')))
+                        
+        json_df[entity_key] = entity_values
+        
+        for key, value in data[1].items():
+            json_df[key] = [getattr(value, value.WhichOneof('val'))]      
+        
+        json_df['event_timestamp'] = [data[2].strftime('%Y-%m-%d %H:%M:%S.%f')]
+        json_df['created'] = [data[3].strftime('%Y-%m-%d %H:%M:%S.%f')]
+        print("json_df", json_df)
+        
+        req_body = {
+                "push_source_name": table.name,
+                "df": json_df,
+                "to": "online"
+            }
+        
+        return req_body
+
+    def _check_if_feature_requested(self, feature_name, requested_features):
+        for requested_feature in requested_features:
+            if feature_name in requested_feature:
```

Comment:
Isn't this a substring check? Shouldn't a simple equality check be sufficient? Or maybe there's an edge case because of `full_feature_names`?

Reply:
Yes, that is correct. Just to incorporate when full_feature_names is used. Please let me know if there is any better way.

Reply:
this returns True if we feature name is a substring of the requested feature, it can return false positives IMO.

Reply:
updated the code after performing some tests, it makes sense to use ==.